docs/apis/calcite.md

```diff
@@ -0,0 +1,570 @@
+
+>   <span style=" font-family: 'Bebas Neue'; font-size:'15'">
+    &#960;<sub>ROWTIME,PRODUCT</sub>&#963;(ORDERS)
+    </span>
+
+Apache Calcite uses relational algebra as basis to create Relation Expression Tree which further used by Apache Apex SQL module to convert functional DAG.Now we will go in to details of how it happens.
+
+But given model configuration and query, code need to read all records from a csv HDFS file and filter out all columns but two from them.  Based on available operators in
+Apache Malhar, requirement for reading from HDFS file line by line and parse each line as csv with provided csv schema is mapped to **_FileInputOperator+CSVParser_**.  For all records parsed correctly by CSVParser, only columns ROWTIME and PRODUCT need to be retained in output.
+This is known as **"PROJECTION" ( &#960; )** in relational algebra. Apache Apex-Calcite integration achieves this by added node in a DAG with **_FilterTranformOperator_**. Since there no output table specified in a query, **_ConsoleOperator_** is used to emit data.
+ 
+
+
+### 2. SQL Application with API:
+
+In previous example we used model file to configure SQLExecution environment . But in this example we will configure environment by configuring input table explicitly through code.
+
+    <configuration>
+
+     <!-- Input Definition -->
+
+     <property>
+
+       <name>csvSchemaInName</name>
+
+       <value>ORDERS</value>
+
+     </property>
+
+     <property>
+
+       <name>csvSchemaIn</name>
+
+       <value>{"separator":",","quoteChar":"\"","fields":[{"name":"RowTime","type":"Date","constraints":{"format":"dd/MM/yyyy hh:mm:ss Z"}},{"name":"id","type":"Integer"},{"name":"Product","type":"String"},{"name":"units","type":"Integer"}]}</value>
+
+     </property>
+
+     <property>
+
+       <name>sourceFile</name>
+
+       <value>/user/hadoop/inputs/input.csv</value>
+
+     </property>
+
+     <!-- Execution SQL -->
+
+     <property>
+
+       <name>sql</name>
+
+       <value>SELECT STREAM ROWTIME, PRODUCT FROM ORDERS</value>
+
+     </property>
+
+    </configuration>
+
+**Configuration file for an application**
+
+ In above configuration file we have explicitly declared properties like csvSchemaInName, csvSchemaIn, sourceFile (input file from HDFS) and sql to be executed .
+
+     @Override
+
+     public void populateDAG(DAG dag, Configuration conf)
+
+     {
+
+       // Source definition
+
+       String schemaInName = conf.get("csvSchemaInName");
+
+       String schemaIn = conf.get("csvSchemaIn");
+
+       String sourceFile = conf.get("sourceFile");
+
+       SQLExecEnvironment.getEnvironment()
+
+           .registerTable(schemaInName, new FileEndpoint(sourceFile, new CSVMessageFormat(schemaIn)))
+
+           .executeSQL(dag, conf.get("sql"));
+
+     }
+
+![image alt text](images/image_1.jpg)
+
+
+** Physical Dag View of Application **
+
+### 3. Pure SQL Style Application with Custom Function:
```

Comment:
I'm not happy with the way the code is explained. This is majorly a copy of what is present in demos/ dir of malhar. We need to explain the API here... No need to put section about conf. The idea here is to explain the SQLExecEnvironment API to user...